### PR TITLE
zboss: Distribute BDB and ZCL layers in sources

### PR DESCRIPTION
**Pull request by tomchy**
- Separate BDB and ZCL layers from the main ZBOSS library
 - Distribute BDB and ZCL layers in source code
 - Move scheduler API prototypes to zb_osif.h
 - Add API for sending and receiving Inter-PAN frames
 - Extend NCP protocol by Inter-PAN commands
 - Add bootloader commands in the NCP protocol

### CODE REVIEW

**Comment by tomchy** [comment]
Testing of the ZBOSS library is currently being finalized. Afterwards - there will be a library and ZBOSS version name update, but the rest of the PR will remain unchanged.

I will remove the DNM label after the ZBOSS update.

**Review by sebastiandraus — APPROVED** [review]

File: zboss/include/osif/build_info.h [hunk 1]
```diff
@@ -11,58 +11,60 @@
...
  * nrf_hw_models      a47e326ca772ddd14cc3b9d4ca30a9ab44ecca16 https://github.com/zephyrproject-rtos/nrf_hw_models
+ * TraceRecorderSource d9889883abb4657d71e15ff055517a9b032f8212 https://github.com/zephyrproject-rtos/TraceRecorderSource
  * edtt               7dd56fc100d79cc45c33d43e7401d1803e26f6e7 https://github.com/zephyrproject-rtos/edtt
+ * tensorflow         dc70a45a7cc12c25726a32cd91b28be59e7bc596 https://github.com/zephyrproject-rtos/tensorflow
  */
 
 
 /** ZBOSS build tag */
-#define ZBOSS_BUILD_TAG "zoi_release-3.5.2.0"
+#define ZBOSS_BUILD_TAG "5dcb873b0"
```
[the remark this answers is not in the document]
Comment (tomchy) [reply zboss/include/osif/build_info.h:58]: Yes, it is not tagged right now.

**Review by rlubos — APPROVED** [review]

File: zboss/include/osif/build_info.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomchy) [reply zboss/include/osif/build_info.h]: Updated

**Comment by tomchy** [comment]
ZBOSS updated. Ready for merges.

File: zboss/CHANGELOG.rst [hunk 2]
```diff
@@ -17,7 +17,12 @@ Changes
 =======
 
 * Reduced the number of ZBOSS libraries to a single set.
-* Updated the ZBOSS stack to version ``3.5.2.0+2.0.0``.
+* Added sources of the BDB and ZCL layers and removed them from the main ZBOSS library
+* Moved scheduler API prototypes to zb_osif.h
```
Comment (b-gent) [inline zboss/CHANGELOG.rst:21]:
```suggestion
* Moved scheduler API prototypes to :file:`zb_osif.h`.
```

File: zboss/CHANGELOG.rst [hunk 3]
```diff
@@ -17,7 +17,12 @@ Changes
...
+* Added API for sending and receiving Inter-PAN frames
+* Extended NCP protocol by Inter-PAN commands
```
Comment (b-gent) [inline zboss/CHANGELOG.rst:23]:
```suggestion
* Extended NCP protocol with inter-PAN commands.
```

File: zboss/CHANGELOG.rst [hunk 4]
```diff
@@ -17,7 +17,12 @@ Changes
 =======
 
 * Reduced the number of ZBOSS libraries to a single set.
-* Updated the ZBOSS stack to version ``3.5.2.0+2.0.0``.
+* Added sources of the BDB and ZCL layers and removed them from the main ZBOSS library
+* Moved scheduler API prototypes to zb_osif.h
+* Added API for sending and receiving Inter-PAN frames
```
Comment (b-gent) [inline zboss/CHANGELOG.rst:22]:
```suggestion
* Added API for sending and receiving inter-PAN frames.
```

File: zboss/CHANGELOG.rst [hunk 5]
```diff
@@ -17,7 +17,12 @@ Changes
...
+* Extended NCP protocol by Inter-PAN commands
+* Added bootloader commands in the NCP protocol
```
Comment (b-gent) [inline zboss/CHANGELOG.rst:24]:
```suggestion
* Added bootloader commands in the NCP protocol.
```

File: zboss/CHANGELOG.rst [hunk 6]
```diff
@@ -17,7 +17,12 @@ Changes
 =======
 
 * Reduced the number of ZBOSS libraries to a single set.
-* Updated the ZBOSS stack to version ``3.5.2.0+2.0.0``.
+* Added sources of the BDB and ZCL layers and removed them from the main ZBOSS library
```
Comment (b-gent) [inline zboss/CHANGELOG.rst:20]:
```suggestion
* Added sources of the BDB and ZCL layers and removed them from the main ZBOSS library.
```

**Review by b-gent — CHANGES_REQUESTED** [review]

File: zboss/CHANGELOG.rst (hunk 2 above)
Comment (tomchy) [reply zboss/CHANGELOG.rst]: Fix applied.

**Review by b-gent — APPROVED** [review]

**Comment by tejlmand** [comment]
@rlubos why was this merged when CI for manifest PR is red ?
image

**Pull request merged**